### PR TITLE
[REVIEW] Bug document

### LINKED ISSUE #393: [BUG] Fix and review 0.9 doc
I suspect that the refactoring removed some features (such as `renumber`) from the nightly doc : https://docs.rapids.ai/api/cugraph/nightly/. The cause is unknown as the content is in the python file and the doc of other files is visible. 

We need to check that all functions are there and re-add the missing ones.

### PR DESCRIPTION
**Pull request by seunghwak**
Close #393 

Fixed various document formatting issues.

### CODE REVIEW

**Comment by afender** [comment]
What's the best way to visualize/render the doc changes contained in this PR?

**Comment by seunghwak** [comment]
> What's the best way to visualize/render the doc changes contained in this PR?

cd cugraph/docs
make html
cd build/html
Open api.html and/or index.html

You can do this in the main repo, then do this with this PR, and compare.

To clone a PR for review, you can do

git fetch origin pull/###/head
git checkout -b pr###_review FETCH_HEAD

Here ### should be PR number (412 in this case).

**Comment by afender** [comment]
Thanks, sorry I should have been more precise. I am aware of this path. It is just that it doesn't render as https://docs.rapids.ai/api/cugraph/stable/ (at least on my machine). Does it look the same on yours?

**Comment by seunghwak** [comment]
> Thanks, sorry I should have been more precise. I am aware of this path. It is just that it doesn't render as https://docs.rapids.ai/api/cugraph/stable/ (at least on my machine). Does it look the same on yours?

Not exactly same (e.g. the online one uses rapids violet and mine on my machine uses bluish colors), and the surrounding frames look somewhat different, but besides that, it looks pretty similar. If you're building this in a remote machine and copying files to your local machine  to see the generated document, you need to copy the entire folder not just the html files for this to work properly.

**Comment by afender** [comment]
I see these warnings when building the doc :
```
/home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/numpydoc/docscrape.py:424: UserWarning: Unknown section Example: in the docstring of <function analyzeClustering_edge_cut at 0x7fed6e6d5268> in /home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cugraph-0.9.0a1+55.g8994b7b-py3.7-linux-x86_64.egg/cugraph/community/spectral_clustering.py.
  warn(msg)
/home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/numpydoc/docscrape.py:424: UserWarning: Unknown section Example: in the docstring of <function analyzeClustering_modularity at 0x7fed6e6d5158> in /home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cugraph-0.9.0a1+55.g8994b7b-py3.7-linux-x86_64.egg/cugraph/community/spectral_clustering.py.
  warn(msg)
/home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/numpydoc/docscrape.py:424: UserWarning: Unknown section Example: in the docstring of <function analyzeClustering_ratio_cut at 0x7fed6e6d52f0> in /home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cugraph-0.9.0a1+55.g8994b7b-py3.7-linux-x86_64.egg/cugraph/community/spectral_clustering.py.
  warn(msg)
/home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/numpydoc/docscrape.py:424: UserWarning: Unknown section Example: in the docstring of <function spectralBalancedCutClustering at 0x7fed6e6d5048> in /home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cugraph-0.9.0a1+55.g8994b7b-py3.7-linux-x86_64.egg/cugraph/community/spectral_clustering.py.
  warn(msg)
/home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/numpydoc/docscrape.py:424: UserWarning: Unknown section Example: in the docstring of <function spectralModularityMaximizationClustering at 0x7fed6e6d50d0> in /home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cugraph-0.9.0a1+55.g8994b7b-py3.7-linux-x86_64.egg/cugraph/community/spectral_clustering.py.
  warn(msg)
/home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/numpydoc/docscrape.py:424: UserWarning: Unknown section Example: in the docstring of <function subgraph at 0x7fed099e2268> in /home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cugraph-0.9.0a1+55.g8994b7b-py3.7-linux-x86_64.egg/cugraph/community/subgraph_extraction.py.
  warn(msg)
/home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/numpydoc/docscrape.py:424: UserWarning: Unknown section Example in the docstring of <function triangles at 0x7fed099e2f28> in /home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cugraph-0.9.0a1+55.g8994b7b-py3.7-linux-x86_64.egg/cugraph/community/triangle_count.py.
  warn(msg)

/home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cugraph-0.9.0a1+55.g8994b7b-py3.7-linux-x86_64.egg/cugraph/structure/graph.py:docstring of cugraph.structure.graph.Graph.degrees:20: WARNING: Definition list ends without a blank line; unexpected unindent.
/home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cugraph-0.9.0a1+55.g8994b7b-py3.7-linux-x86_64.egg/cugraph/structure/convert_matrix.py:docstring of cugraph.structure.convert_matrix.from_cudf_edgelist:8: WARNING: Unexpected indentation.
/home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cugraph-0.9.0a1+55.g8994b7b-py3.7-linux-x86_64.egg/cugraph/structure/convert_matrix.py:docstring of cugraph.structure.convert_matrix.from_cudf_edgelist:11: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cugraph-0.9.0a1+55.g8994b7b-py3.7-linux-x86_64.egg/cugraph/link_analysis/pagerank.py:docstring of cugraph.link_analysis.pagerank.pagerank:26: WARNING: Unexpected indentation.
/home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cugraph-0.9.0a1+55.g8994b7b-py3.7-linux-x86_64.egg/cugraph/link_analysis/pagerank.py:docstring of cugraph.link_analysis.pagerank.pagerank:27: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cugraph-0.9.0a1+55.g8994b7b-py3.7-linux-x86_64.egg/cugraph/link_analysis/pagerank.py:docstring of cugraph.link_analysis.pagerank.pagerank:48: WARNING: Unexpected indentation.
/home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cugraph-0.9.0a1+55.g8994b7b-py3.7-linux-x86_64.egg/cugraph/link_analysis/pagerank.py:docstring of cugraph.link_analysis.pagerank.pagerank:49: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cugraph-0.9.0a1+55.g8994b7b-py3.7-linux-x86_64.egg/cugraph/link_prediction/wjaccard.py:docstring of cugraph.link_prediction.wjaccard.jaccard_w:41: WARNING: Unexpected indentation.
/home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cugraph-0.9.0a1+55.g8994b7b-py3.7-linux-x86_64.egg/cugraph/link_prediction/wjaccard.py:docstring of cugraph.link_prediction.wjaccard.jaccard_w:46: WARNING: Unexpected section title or transition.
/home/afender/anaconda3/envs/cugraph_dev/lib/python3.7/site-packages/cugraph-0.9.0a1+55.g8994b7b-py3.7-linux-x86_64.egg/cugraph/link_prediction/woverlap.py:docstring of cugraph.link_prediction.woverlap.overlap_w:42: WARNING: Unexpected indentation.
```

**Comment by afender** [comment]
There was a discussion yesterday on how doc example should show how to use the API instead of being a fully working code. The goal is that the user understands the feature and notebooks provide working (tested) code.
I think examples for the graph structure should be updated.

**Comment by afender** [comment]
Screenshot from 2019-07-30 12-01-33

**Comment by afender** [comment]
Wrong formating in return part for the second DF
Screenshot from 2019-07-30 12-02-40

**Comment by afender** [comment]
"Specified" is not properly colored (there are many of those)
Screenshot from 2019-07-30 12-02-40
)

**Comment by afender** [comment]
Can we update all examples to use cudf csv reader instead of the matrix market reader?

**Comment by afender** [comment]
I deleted a few outdated comments after realizing that the python package should be (re)built prior to building the doc. Let's update the documentation readme to mention this step.

**Comment by seunghwak** [comment]
@afender We need to rebuild and reinstall python package before compiling documents. "make html" works on the installed cugraph package.

```
(cuda10) seunghwak@seunghwak-linux:~/RAPIDS/cugraph/docs$ make html
Running Sphinx v2.1.2
making output directory... done
loading intersphinx inventory from https://docs.python.org/objects.inv...
intersphinx inventory has moved: https://docs.python.org/objects.inv -> https://docs.python.org/3/objects.inv
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 2 source files that are out of date
updating environment: 2 added, 0 changed, 0 removed
reading sources... [ 50%] api
reading sources... [100%] index

looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [ 50%] api
writing output... [100%] index

generating indices... genindex py-modindex
writing additional pages... search/home/seunghwak/Program/anaconda3/envs/cuda10/lib/python3.7/site-packages/sphinx_rtd_theme/search.html:20: RemovedInSphinx30Warning: To modify script_files in the theme is deprecated. Please insert a <script> tag directly in your theme instead.
  {{ super() }}

copying static files... done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded.
```

Then, the output log should look like the above.

There is still one remaining warning, but it's due to an external package (sphinx_rtd_theme). It seems like it is already fixed (https://github.com/readthedocs/sphinx_rtd_theme/issues/739) on March 28th, but is not incorporated in the version we are using (0.4.3, the last commit was on Feb. 12th).

**Comment by seunghwak** [comment]
> Can we update all examples to use cudf csv reader instead of the matrix market reader?

Regarding your requests on examples, I can do either in this PR or another PR to close https://github.com/rapidsai/cugraph/issues/372. My original plan was to do this in a separate PR to close #372, but either is fine for me.

**Comment by afender** [comment]
Nice job! The doc looks much better overall. 
The comments above capture doc related issues that we discussed separately. This PR is a great occasion to address them (it should be quick additions, let me know if that's not the case).

File: python/cugraph/structure/graph.py [hunk 1]
```diff
@@ -177,17 +177,17 @@ def delete_edge_list(self):
...
-        represent a graph using the adjacency list format. If value_col is
-        None, an unweighted graph is created. If value_col is not None, an
-        weighted graph is created. If copy is False, this function stores
-        references to the passed objects pointed by offset_col and index_col.
-        If copy is True, this funcion stores references to the deep-copies of
-        the passed objects pointed by offset_col and index_col. If this class
-        instance already stores a graph, invoking this function raises an
-        error. Undirected edges must be stored as directed edges in both
-        directions.
+        Initialize a graph from the adjacency list. It is error to call this
```
Comment (kaatish) [inline python/cugraph/structure/graph.py]: 'It is error' -> 'It is erroneous'.

File: python/cugraph/structure/graph.py [hunk 2]
```diff
@@ -306,23 +307,24 @@ def delete_transposed_adj_list(self):
 
     def get_two_hop_neighbors(self):
         """
-        Return a dataframe containing vertex pairs such that each pair of
-        vertices is connected by a path of two hops in the graph. The resulting
-        pairs are returned in sorted order.
+        Compute vertex pairs that are two hop apart. The resulting pairs are
```
Comment (kaatish) [inline python/cugraph/structure/graph.py]: hop -> hops sounds better.

File: python/cugraph/structure/graph.py [hunk 3]
```diff
@@ -63,24 +64,23 @@ def clear(self):
 
     def add_edge_list(self, source_col, dest_col, value_col=None, copy=False):
         """
-        Create the edge list representation of a Graph. The passed source_col
-        and dest_col arguments wrap gdf_column objects that represent a graph
+        Initialize a graph from the edge list. It is error to call this
```
Comment (kaatish) [inline python/cugraph/structure/graph.py]: 'It is an error'

**Comment by afender** [comment]
@kaatish please help us with a second review of the doc. I'd like the doc to be great for 0.9 release.

File: python/cugraph/structure/graph.py (hunk 1 above)
Comment (seunghwak) [reply python/cugraph/structure/graph.py]: Thanks for finding these typos, I made a fix.

**Comment by seunghwak** [comment]
@afender @kaatish I made the requested fixes (and more).

**Pull request merged**